.github/workflows/build.yaml

```diff
@@ -20,30 +20,68 @@ on:
     - cron: '0 3 * * *'
 
 jobs:
+
   build_and_push:
+
     name: Build ruby_${{ matrix.version }} images and push to GHCR
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        version: ['3_1', '3_1_2', '3_2', '3_2_0']
+        version:
```

Comment:
Only need to build each version once and tag twice ; made the version numbers JSON so GHA can manipulate them easily, added a flag for "this version gets to be tagged latest".